django/template/base.py

```diff
@@ -805,7 +809,7 @@ def __init__(self, var):
             except ValueError:
                 # Otherwise we'll set self.lookups so that resolve() knows we're
                 # dealing with a bonafide variable
-                if var.find(VARIABLE_ATTRIBUTE_SEPARATOR + '_') > -1 or var[0] == '_':
+                if VARIABLE_ATTRIBUTE_SEPARATOR + '_' in var or first_char == '_':
```

Comment:
We update `var` in the
https://github.com/django/django/blob/f03ba0ad52e6f1deb42f022090f226573b22f630/django/template/base.py#L800
so I'd revert to the `var[0]`.

Reply:
Yeah I agree, the best course of action is to remove `first_char` and `last_char` entirely, and just use `var[0]` and `var[-1]` in their place. Should be (nearly)the same because the minor improvement was from removing `startswith` and `endswith`. 
Still weird that the tests only sometimes failed though, right?